
[tox]
envlist = py{37,38,39,310}-django{22,32}, py37-flake8, py37-coverage

[testenv:py37-flake8]
deps = flake8
commands = flake8

[testenv:py37-coverage]
whitelist_externals = echo
commands =
    pip install coverage
    coverage erase
    coverage run ./manage.py test
    coverage report --include='admin_sort*' --omit='*/tests/*'
    coverage html --include='admin_sort*' --omit='*/tests/*'
    echo "opener htmlcov/index.html"

[testenv]
commands = python manage.py test
setenv =
    DJANGO_SETTINGS_MODULE=admin_sort.tests.settings
    PYTHONPATH={toxinidir}
deps =
    django22: Django>=2.2,<2.3
    django32: Django>=3.2,<4
    django-appconf
    selenium<4.0.0
# https://stackoverflow.com/questions/70821737/webdriverexception-message-service-geckodriver-unexpectedly-exited-status-cod

[testenv:py27-django20]
whitelist_externals = echo
deps =
commands= echo "no django 2.0 on python 2.7!"

[testenv:py27-django21]
whitelist_externals = echo
deps =
commands= echo "no django 2.1 on python 2.7!"

[testenv:py27-django22]
whitelist_externals = echo
deps =
commands= echo "no django 2.2 on python 2.7!"